### PR TITLE
Input Interaction: fix cmd+arrow

### LINKED ISSUE #5805: Cmd [arrow] commands do not function as expected
## Issue Overview
In macOS, <kbd>⌘ [arrow]</kbd> commands have well-defined meanings, in a text-editing context.

- <kbd>⌘ &uarr;</kbd> takes your caret to the beginning of your content (top-left in LTR languages).
- <kbd>⌘ &darr;</kbd> takes your caret to the end of your content (bottom-right in LTR languages).
- <kbd>⌘ &larr;</kbd> takes your caret to the beginning of the current line.
- <kbd>⌘ &rarr;</kbd> takes your caret to the end of the current line.

In Gutenberg, these commands inconsistently navigate your caret between blocks.

- <kbd>⌘ &uarr;</kbd> takes your caret to the same x-value in the previous block.
- <kbd>⌘ &uarr;</kbd> **in the first block** takes your caret to the end of the title.
- <kbd>⌘ &darr;</kbd> takes your caret to the same x-value in the next block.
- <kbd>⌘ &rarr;</kbd> **in the first block** takes your caret to the start of the next block.
- <kbd>⌘ &rarr;</kbd> **in a middle block** takes your caret to the end of that same block (this is correct!)
- <kbd>⌘ &rarr;</kbd> **in an ending block** takes your caret to the start of a new block.
- <kbd>⌘ &larr;</kbd> takes your caret to the end of the previous block.

## Expected Behavior

macOS default behavior.

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Fixes #5805 and #8353.

* For cmd + horizontal arrow: always fall back to browser behaviour.
* For cmd + arrow up: select the first block.
* For cmd + arrow down: select the last block.
* For ctrl + home: select the first block.
* For ctrl + end: select the last block.

Question: does ctrl + arrow work the same outside macOS?

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by ellatrix** [comment]
@Clorith and @mrwweb: since you raised the issue of ctrl + home/end (and assumably don't use macOS), do you know what ctrl + up/down/left/right does normally in a text field (have a look at [TinyMCE](https://www.tiny.cloud) or a plain `textarea`)?

File: packages/block-editor/src/components/writing-flow/index.js [hunk 1]
```diff
@@ -269,6 +273,18 @@ class WritingFlow extends Component {
 			return;
 		}
 
+		if ( metaKey || ctrlKey ) {
+			if ( ( metaKey && isUp ) || ( ctrlKey && keyCode === HOME ) ) {
```
Comment (aduth) [inline packages/block-editor/src/components/writing-flow/index.js]: Is Ctrl+Home OS-specific? It doesn't do anything for me in other native inputs (this textarea on GitHub, a Google Docs document, a Sublime Text file), nor does any other combination of modifier + Home. In which case, we have the reverse problem that we're handling a combination we shouldn't be?

Note: While most standard Mac laptops won't have a physical Home key, in my case I connect by an [external keyboard](https://www.logitechg.com/en-us/products/gaming-keyboards/pro-gaming-keyboard.html) which does.

**Comment by paaljoachim** [comment]
I am checking in. 
Is this PR still valid?

**Comment by Clorith** [comment]
> @Clorith and @mrwweb: since you raised the issue of ctrl + home/end (and assumably don't use macOS), do you know what ctrl + up/down/left/right does normally in a text field (have a look at [TinyMCE](https://www.tiny.cloud) or a plain `textarea`)?

You're right that I'm on a Windows machine.

ctrl+left/right jumps between either full words strings (or between stop-symbols, for example `/` or `|`, if they are used in the middle of a connected string), in the designated direction.

I don't believe ctrl+up/down does anything special, at least not that I've noticed.

**Comment by kathrynwp** [comment]
Hey there @ellatrix @Clorith - since there hasn't been any movement on this one in a couple of years, just checking in one last time to see whether this is still valid, or if it can be closed. Thanks for having a peek when you're able.

**Comment by paaljoachim** [comment]
I believe this issue is associated...
Tracking: Some keyboard shortcuts don't work on Windows
https://github.com/WordPress/gutenberg/issues/43352

**Comment by Clorith** [comment]
Can confirm that <kbd>CTRL</kbd>+<kbd>HOME</kbd> and <kbd>CTRL</kbd>+<kbd>END</kbd> on Windows still only works within the selected block, and has the same behavior as <kbd>CTRL</kbd>+<kbd>UP</kbd> and <kbd>CTRL</kbd>+<kbd>DOWN</kbd> respectively, in that they only jump to the start/end of the block.

The use of <kbd>UP</kbd> and <kbd>DOWN</kbd> to move to the start or end of a block/paragraph is specific to the block editor though, and is not native Windows functionality.

**Comment by t-hamano** [comment]
> Can confirm that CTRL+HOME and CTRL+END on Windows still only works within the selected block, and has the same behavior as CTRL+UP and CTRL+DOWN respectively, in that they only jump to the start/end of the block.

I also tested it with the latest Gutenberg and it behaves the same way as above on Windows.

I tried to reflect the changes made in this PR in the latest trunk, but this file is separated as [use-arrow-nav.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/writing-flow/use-arrow-nav.js) and there is a lot of refactoring going on and I could not figure out how to reflect it.

However, I feel that the logic added in this PR is valid for Windows in the latest Gutenberg.

**Pull request closed without merging**